File: hrms-frontend/src/services/grievanceService.js
// src/services/grievanceService.js
import api from '../api/axiosConfig';

const grievanceService = {
  getAll: () => api.get('/grievances'),
  getById: (id) => api.get(`/grievances/${id}`),
  getByEmployee: (empId) => api.get(`/grievances/employee/${empId}`),
  getByStatus: (status) => api.get(`/grievances/status/${status}`),
  getMyAssigned: () => api.get('/grievances/assigned/me'),
  create: (data) => api.post('/grievances', data),

  updateStatus: (id, status) =>
    api.patch(`/grievances/${id}/status?status=${status}`),
  assign: (id, userId) =>
    api.patch(`/grievances/${id}/assign`, { userId }),
  resolve: (id, resolution) =>
    api.patch(`/grievances/${id}/resolve`, { resolution }),
  escalate: (id) => api.patch(`/grievances/${id}/escalate`),
  submitFeedback: (id, data) =>
    api.patch(`/grievances/${id}/feedback`, data),

  addComment: (id, data) => api.post(`/grievances/${id}/comments`, data),
  getComments: (id) => api.get(`/grievances/${id}/comments`),

  getStatistics: () => api.get('/grievances/statistics'),
};

export default grievanceService;

File: hrms-backend/src/main/java/com/hrms/service/GrievanceService.java

//service/GrievanceService.java
package com.hrms.service;

import com.hrms.dto.response.GrievanceResponse;
import com.hrms.model.Grievance;
import com.hrms.model.GrievanceComment;

import java.util.List;
import java.util.Map;

public interface GrievanceService {
 GrievanceResponse create(Grievance grievance);
 GrievanceResponse getById(Long id);
 List<GrievanceResponse> getAll();
 List<GrievanceResponse> getByEmployee(Long employeeId);
 List<GrievanceResponse> getByStatus(String status);
 List<GrievanceResponse> getMyAssigned(Long userId);

 GrievanceResponse updateStatus(Long id, String status);
 GrievanceResponse assign(Long id, Long userId);
 GrievanceResponse resolve(Long id, String resolution, Long userId);
 GrievanceResponse escalate(Long id);
 GrievanceResponse submitFeedback(Long id, Integer rating, String feedback);

 GrievanceComment addComment(Long grievanceId, Long userId,
                               String comment, Boolean isInternal);
 List<GrievanceComment> getComments(Long grievanceId);

 Map<String, Object> getStatistics();
}